coriolis_openstack_utils/cli/router.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from oslo_log import log as logging
+
+from cliff import lister
+
+from coriolis_openstack_utils import conf
+from coriolis_openstack_utils.actions import network_actions
+from coriolis_openstack_utils.cli import formatter
+from coriolis_openstack_utils.resource_utils import routers
+
+
+LOG = logging.getLogger(__name__)
+
+
+class RouterMigrationFormatter(formatter.EntityFormatter):
+    columns = (
+        "Router Name",
+        "Router ID",
+        "Tenant ID")
+
+    def _get_formatted_data(self, obj):
+        data = (
+            obj["name"],
+            obj["id"],
+            obj["tenant_id"])
+
+        return data
+
+
+class MigrateRouter(lister.Lister):
+    def get_parser(self, prog_name):
+        parser = super(MigrateRouter, self).get_parser(prog_name)
+        source_group = parser.add_mutually_exclusive_group(required=True)
+        source_group.add_argument(
+            "--source-router-id",
+            dest="src_router_id",
+            help="The id of the router that is being migrated.")
+        source_group.add_argument(
+            "--source-router-name", dest="src_router_name",
+            help="The name of the router that is being "
+                 "migrated.")
+        parser.add_argument(
+            "--not-a-drill", dest="not_drill", action="store_true",
+            default=False,
+            help="If unset, tooling will only print the indented operations.")
+
+        return parser
+
+    def take_action(self, args):
+        # instantiate all clients:
+        source_client = conf.get_source_openstack_client()
+        destination_client = conf.get_destination_openstack_client()
+
+        if args.src_router_name:
+            src_router_id = routers.get_router(
+                source_client, args.src_router_name)['id']
+        else:
+            src_router_id = args.src_router_id
+
+        router_creation_payload = {
+            'src_router_id': src_router_id}
+
+        router_creation_action = network_actions.RouterCreationAction(
+            source_client, destination_client, router_creation_payload)
+
+        done = router_creation_action.check_already_done()
+        router = []
```

Comment:
Considering you're constructing the list with the router (they'll only ever be one) in the formatter call on line 85, please initialize this to `None`